### ISSUE TITLE
Using --global-option in requirements.in causes attribute error

### ISSUE THREAD

**Issue by rrauenza** [jazzband/pip-tools#345]
In order to compile this package (or even check its dependencies via pip-compile), it needs options passed to setup.py.  Pip supports passing --install-option and --global-option in the requirements.txt, but this seems to cause problems with pip-compile:

requirements.in:

```
p4python==2014.1.831633 --global-option="--apidir=/tmp/p4api-2014.1.821990"
```

Output:

```
$ bin/pip-compile           
▽raceback (most recent call last):
  File "/opt/virtualenv-2.7.10/bin/pip-compile", line 11, in <module>
    sys.exit(cli())
  File "/opt/virtualenv-2.7.10/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/opt/virtualenv-2.7.10/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/opt/virtualenv-2.7.10/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/virtualenv-2.7.10/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/opt/virtualenv-2.7.10/lib/python2.7/site-packages/piptools/scripts/compile.py", line 158, in cli
    src_file, finder=repository.finder, session=repository.session, options=pip_options))
  File "/opt/virtualenv-2.7.10/lib/python2.7/site-packages/pip/req/req_file.py", line 93, in parse_requirements
    for req in req_iter:
  File "/opt/virtualenv-2.7.10/lib/python2.7/site-packages/pip/req/req_file.py", line 150, in process_line
    cmdoptions.check_install_build_global(options, opts)
  File "/opt/virtualenv-2.7.10/lib/python2.7/site-packages/pip/cmdoptions.py", line 56, in check_install_build_global
    control = options.format_control
AttributeError: Values instance has no attribute 'format_control'
```

format_control appears to be an option related to --no-binary or --only-binary, and I'm wondering if the option is getting lost/left out in your own handling of the pip command line within pip-compile.

**Comment by jimbocoder** [comment]
Also running into this one. @rrauenza did you come up with any nice workarounds when you encountered this issue?

**Comment by rrauenza** [comment]
wow -- it's been so long, I don't recall if or how I worked around this. Sorry :(

**Comment by awaisdar001** [comment]
Upgrading my pip version helped me out!!!

**Comment by atugushev** [comment]
It's fixed since `pip-tools==3.0.0`.
